index.js

```diff
@@ -849,24 +850,29 @@ RingPop.prototype.handleOrProxyAll =
         }
 
         dests.forEach(function(dest) {
+            var destKeys = keysByDest[dest];
             var res = hammock.Response(function(err, resp) {
                 onResponse(err, resp, dest);
             });
             if (whoami === dest) {
                 self.logger.trace('handleOrProxyAll was handled', {
-                    keys: keys,
+                    keys: destKeys,
```

Comment:
It wasn't a bug in the strictest sense of the word, but it's certainly more useful now than it was before